table/tables/tables.go

```diff
@@ -432,6 +433,12 @@ func (t *tableCommon) AddRecord(ctx sessionctx.Context, r []types.Datum, skipHan
 		if err != nil {
 			return 0, errors.Trace(err)
 		}
+	} else if hasExplicitRowID {
+		// if _tidb_rowid is explicitly assigned, we need to rebase the allocator
+		err = t.RebaseAutoID(ctx, recordID, false)
```

Comment:
RebaseAutoID maybe failed because https://github.com/pingcap/tidb/blob/release-2.1/meta/autoid/autoid.go#L111
And _tidb_rowid allocation is not continuously increasing if there are more than 1 tidb.  This may need to be carefully considered.